homeassistant/components/google_mail/__init__.py

```diff
@@ -37,8 +41,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     auth = AsyncConfigEntryAuth(session)
     try:
         await auth.check_and_refresh_token()
-    except ClientResponseError as err:
-        if 400 <= err.status < 500:
+    except HomeAssistantError as err:
+        resp = err.args[0]
+        if isinstance(resp, ClientResponseError) and 400 <= resp.status < 500:
```

Comment:
Do this check in the api module instead and raise `ConfigEntryAuthFailed` from there. It's a subclass of `HomeAssistantError`.